docs/Community/Governance.md

```diff
@@ -0,0 +1,137 @@
+
+Technical Discussions
+---------------------
+
+Technical discussions generally happen in one of these places:
+
+* [on the GitHub project](https://github.com/openenclave/openenclave), either in
+  an Issue or a Pull Request
+* in the public meetings of Committees, Special Interest Groups, or Working Groups, which can be found via the [community calendar](https://lists.confidentialcomputing.io/calendar).
+* on the community mailing list or chat server, which is detailed in the [Comminucations](Communication.md) document
+
+While we strive for openness and transparency, it is natural that some
+discussions will also happen in private channels, or in impromptu face-to-face
+conversations (for those who work in the same building, or bump into each other
+at a conference). The outcome of such discussions should be considered
+non-binding until the information is shared through an open medium and relevant
+stake-holders have opportunity to contribute to the discussion.
+
+All SIG and WG leaders are responsible for creating and publishing minutes from
+the meetings they hold, ensuring that the meetings are open and inclusive, and
+that decisions they make are fair and transparent.
+
+If a private discussion pertains to an open issue or PR, and results in a
+decision, that item should be updated and reference given to the conversation.
+
+Issue/PR Review Process
+-----------------------
+
+The community strives to triage and respond to all incoming issues within one
+week. If you haven't heard anything by then, or an issue or PR is not getting
+any response within a reasonable time after it is triaged, please feel free to
+remind us with a ping on the thread.
+
+Security issues should be reported through a separate channel, and
+will receive a response within 24 hours. See [Reporting Security
+Issues](Contributing.md#reporting-security-issues).
+
+Committers and Contributors
+---------------------------
+
+At present, we maintain a list of [Project Committers](Committers.md) along with
+descriptions of their knowledge areas.
+
+This will be updated and replaced with a more scaleable approach to
+commit-rights management.
+
+Accepting Contributions
+-----------------------
+
+SIGs will generally accept changes that improve the project and that align with the
+[Open Enclave roadmap](https://github.com/openenclave/openenclave/projects).
+
+All contributions are expected to be reviewed by at least one SIG lead other
+than the contributor, and have no significant objections from other SIG leads
+within the same SIG.
+
+If a change requires the approval of multiple SIGs (e.g., because it affects
+code in both SIG domains), then approval from both SIGs is required.
+
+If a change only requires one approval from one SIG, and leads of another SIG
+object to it, a discussion should be invited and given reasonable time to move
+towards a consensus; overruling the objection of another SIG lead should only be
```

Comment:
* Charter.md still has old text that suggests that the CGC is still an arbitration body of last resort for _technical_ decisions as well.

   >This Committee is not intended to make every technical decision, as those should
generally be made by agreement among committers as PRs are reviewed and merged.
Where disagreements take place and need further resolution, those can be brought
up with the Committee as part of its responsibility to maintain the project's
community health. Otherwise technical decisions are left to the active
committers (by virtue of the liberal contribution policy).

   Is that still accurate in the new governance structure, or should that be revised and made clear that SIGs are responsible for arbitration of technical issues?

* Is there an expectation that all SIGs are co-equal in this regard? For example, I can imagine that SIG-architecture will need to drive decisions across SIGs regularly, and how do those relationships work?

* Other projects have the notion of a technical steering committee, which do serve as a final body for technical arbitration. In the proposed taxonomy, would that be like a kind of permanent WG with representation from all SIGs? Is that a useful concept?

Reply:
Great question ... yeah, I could revise this section further.

Every other projects I've been involved in had a "technical committee" of some variety, whether it was called the TC / SC / TAC / TOC, or the project was led by a BDFL (*). That committee (or individual) also determined the governance structure, resolved disputes between projects or SIGs, and some of them also put effort into the community health.

I don't know that unifying both technical and community leadership in one body is better than separating them, especially at an early stage. My suggestion would be to keep them unified right now. Perhaps that committee is made up of the SIG leads? *shrug*

Oh, and Kubernetes also has a "conduct committee", though that is really an escalation point for serious conduct issues or consent incidents, which local leads often aren't trained to handle.

(*) acronym decoder:
- TC: openstack [technical committee](https://governance.openstack.org/tc/)
- SC: kubernetes [steering committee](https://github.com/kubernetes/steering)
- TOC: CNCF [technical oversight committee](https://github.com/cncf/toc)
- TAC: CCC [technical advisory committee](https://github.com/confidential-computing/governance/blob/master/scoping.md) ... not the greatest docs, but also a much younger body
- BDFL: benevolent dictator for life (eg., Linus Torvalds, Guido van Rossum, ...)

Reply:
As an aside, in this proposal I am seeking to address current development and review scaling challenges by proposing a system for delegating responsibility and improving the github workflow. I did not want to include a proposal for a new structure for the CGC, or for a technical governance committee, because I know that I don't know enough about that layer yet.

When I do my next iteration of this changeset, I'll see if there are ways I can scope the changes more narrowly.

Reply:
> I did not want to include a proposal for a new structure for the CGC, or for a technical governance committee, because I know that I don't know enough about that layer yet.

I agree that it is not necessary to restructure the CGC or its responsibilities towards/away from technical/steering committees used in other open source projects as part of this PR. My last bullet was really a guess at whether you were suggesting going down that road. Returning to the scoped concerns:

> - Is that still accurate in the new governance structure, or should that be revised and made clear that SIGs are responsible for arbitration of technical issues?

It sounds like the intention is that if SIGs do not come to an agreement, then yes, the CGC would be responsible for arbitrating technical disagreements for now. If so, that should probably be stated in the text here; The statement that SIG leads should not overrule each other does not currently indicate where the process bottoms out, and I think that would be helpful to include.

> - Is there an expectation that all SIGs are co-equal in this regard? For example, I can imagine that SIG-architecture will need to drive decisions across SIGs regularly, and how do those relationships work?

I'm not sure I have clarity on this question. Elsewhere in the PR discussions, it seems to be suggested that SIGs are really divided along lines of directory ownership for automation purposes, so the SIG-architecture would be defined by its ownership of docs/DesignDocs. I think some additional elaboration about how the SIGs relate to one another would be helpful here. 

For example, if we don't want to decide to establish a technical steering committee at this point, it seems equally useful to state that SIG leads from other SIGs should also be included in SIG-architecture because it's pretty much everyone's responsibility to review design documents in docs/DesignDocs.

Reply:
> For example, if we don't want to decide to establish a technical steering committee at this point, it seems equally useful to state that SIG leads from other SIGs should also be included in SIG-architecture because it's pretty much everyone's responsibility to review design documents in docs/DesignDocs.

Hmm... what do you think about a structure where all sig leads are "reviewers" in the docs/DesignDocs folder, and the sig-arch lead(s) function is to facilitate the discussion to reach consensus before approving any change? Basically, sig-arch lead is a meeting chair that makes sure all the right people are included in the review process for any given doc before approving it.

Does that help clarify my thinking on this? 

It may be that we need to establish a technical steering committee at the same time as SIGs, and that's OK. I think your points about having a clear resolution path for cross-sig disagreements is completely valid and I'm not sure of the best way to address it today.

Reply:
>Hmm... what do you think about a structure where all sig leads are "reviewers" in the docs/DesignDocs folder, and the sig-arch lead(s) function is to facilitate the discussion to reach consensus before approving any change? Basically, sig-arch lead is a meeting chair that makes sure all the right people are included in the review process for any given doc before approving it.

Yeah, I think that does help clarify the approach and would be helpful to include in the docs. Thanks for working through my questions! 

Regarding establishing a technical steering committee at the same time as the SIGs, I don't have a strong opinion either way; I think what I'm mostly looking for is something in the governance that describes the expectations around how SIGs coordinate and I think a TSC is kind of a common way to deal with that. I think this discussion provides the kind of touchpoints I was looking for either way:

- I think it's really valuable if a process provides guidance for resolving disagreements and deadlocks as a backstop, so an escalation to CGC as previously proposed does address that.
- I think it's better still if governance builds in coordination mechanisms to reduce the occurrence of such issues. For example, I think of the clarification you provided for how SIG leads review the DesignDocs folder as being valuable in that regard.